components/pages/Home/Home.tsx

```diff
@@ -10,19 +11,34 @@ import {
   Footer,
 } from "components";
 import styled from "styled-components";
+import { useIntersectionObserver } from "hooks";
 
 export function Home() {
+  const headerRef = useRef<HTMLDivElement | null>(null);
+  const howItWorksRef = useRef<HTMLDivElement | null>(null);
+  const eHeader = useIntersectionObserver(headerRef, {
+    threshold: 0.1328,
+  });
+  const isIntersectingLightArea = !!eHeader?.isIntersecting;
+  const eHotItWorks = useIntersectionObserver(howItWorksRef, {
+    threshold: 0.49,
+  });
+  const isIntersectingHowItWorksSection = !!eHotItWorks?.isIntersecting;
   return (
     <Layout>
       <Wrapper>
-        <Header />
+        <Header isIntersecting={isIntersectingLightArea} activeLink={isIntersectingHowItWorksSection ? 0 : -1} />
         <Hero />
```

Comment:
Question I have here: creating 5 refs to determine active seems a bit hacky. Is there a better way of doing this in a cleaner manner?